src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -0,0 +1,116 @@
+	public void loadRegistrationSummaryPage() {
+		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage
+				.goToRegistrationSummary().waitForPage();
+	}
+
+	@Then("the system loads Registration Summary Page")
+	public void systemloadsRegistrationEditPage() {
+		assertTrue(registrationSummaryPage.containsText("DEMOGRAPHICS"));
+	}
+
+	@When("a user clicks on Edit link on Registration Summary Page")
+	public void loadDemographicsEditSection() {
+		registrationEditSectionPage = (RegistrationEditSectionPage) dashboardPage
+				.clickOnEditPatient().waitForPage();
+	}
+
+	@And("the system loads the edit demographics section")
+	public void systemloadsEditDemographicSection() {
+		assertTrue(textExists("Demographics"));
+	}
+
+	@When("a user enters the demographics of the patient")
+	public void captureNewDemographics() {
+		registrationEditSectionPage.enterGivenName("Jane");
+		registrationEditSectionPage.enterMiddleName("Adyeri");
+		registrationEditSectionPage.enterFamilyName("Namata");
+		registrationEditSectionPage.selectPatientGender("Male");
+		registrationEditSectionPage.clickOnBirthdateLabel();
+		registrationEditSectionPage.enterBirthDay("23");
+		registrationEditSectionPage.selectBirthMonth("May");
+		registrationEditSectionPage.enterBirthYear("1990");
+	}
+
+	@Then("the system saves the updated patient demographics")
+	public void systemSavesUpdateDemographics() {
+		assertTrue(textExists("DEMOGRAPHICS"));
+	}
+
+	@When("a user clicks on Edit link under contact info section")
+	public void loadContactInfoEditSection() {
+		dashboardPage.clickOnShowContact();
+		registrationEditSectionPage = (RegistrationEditSectionPage) dashboardPage
+				.clickOnEditContact().waitForPage();
+	}
+
+	@And("the system loads the edit contact information section")
+	public void systemloadsEditContactInfoSection() {
+		assertTrue(textExists("Contact Info"));
+	}
+
+	@When("a user enters the contact information of the patient")
+	public void captureNewContactInfo() {
+		registrationEditSectionPage.clearVillage();
+		registrationEditSectionPage.enterVillage("Kyambogo");
+		registrationEditSectionPage.clearState();
+		registrationEditSectionPage.enterState("Central");
+		registrationEditSectionPage.clearCountry();
+		registrationEditSectionPage.enterCountry("Uganda");
+		registrationEditSectionPage.clearPostalCode();
+		registrationEditSectionPage.enterPostalCode("256");
+		registrationEditSectionPage.clickOnPhoneNumberEdit();
+		registrationEditSectionPage.enterPhoneNumber("0785456654");
+	}
+
+	@And("a user clicks on the confirm button")
+	public void saveUpdatedContactInfo() throws InterruptedException {
+		registrationEditSectionPage.clickOnConfirmEdit();
+		dashboardPage = registrationEditSectionPage.confirmPatient();
+	}
+
+	@Then("the system saves the updated patient information")
+	public void systemSavesUpdateContactInfo() {
+		assertTrue(textExists("CONTACT INFO"));
```

Comment:
Does the text `CONTACT INFO` appear only after saving the contact information?

Reply:
@kdaud  after saving the contact info the test goes back to the  Registration Summary Page where CONTACT INFO is visiible